File: src/services/reading-time.js
const averageWordsPerMinute = 200

/**
 * Counts the number of words inside a given post.
 *
 * @param {*} postContent to count words from.
 */
export const countWords = postContent => {
    return postContent.split(/\W/g).filter(c => c).length
}

/**
 * Calculates the number of minutes it'd take to read the given post.
 *
 * @param {*} postContent input to calculate.
 */
export const minutesToRead = postContent => {
    return countWords(postContent) / averageWordsPerMinute
}

export default { countWords, minutesToRead }

File: src/Readder.Core/ReadingTime.fs
module Readder.Core.ReadingTime

open Readder.Core.Types

let private countWords post = post.Content.Split(' ').Length

let private averageWordsPerMinute = 200

/// Calculates an estimate of how much it would take to read the given post
/// based on the average person's words per minute.
let minutesToRead post = countWords post / averageWordsPerMinute
